api/envoy/config/route/v3/route_components.proto

```diff
@@ -388,6 +388,22 @@ message RouteMatch {
     google.protobuf.BoolValue validated = 2;
   }
 
+  // [#not-implemented-hide:]
+  // Configuration for sending data upstream as a raw data payload. This is used for
+  // CONNECT requests, when forwarding CONNECT payload as raw TCP.
+  message ProxyConfig {
+    // TODO(alyssawilk) add proxy proto configuration here.
```

Comment:
also @htuch as long as you're doing an API pass, this Proxyconfig would include (potentially among other things)  the ProxyProtocol message from over on https://github.com/envoyproxy/envoy/pull/10682/files

Basically we're treating the HTTP/2 data stream as sockets, so we'd need to be able to write the proxy header as a payload prefix on individual streams.  Given we're kind of treating it as a socket, we could arguably leave it in its current location, but it felt a little weird to have the HTTP code pull transport socket API protos.

Reply:
I assume this is just the version part of that message, and the nested transport socket goes away. That seems totally reasonable.

Reply:
If it were just the enum (which it is today!) I'd just copy paste.  I guess if we ever wanted to support extensions (coped from Greg's comment on https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) it'd be good to share config

        #define PP2_TYPE_ALPN           0x01
        #define PP2_TYPE_AUTHORITY      0x02
        #define PP2_TYPE_CRC32C         0x03
        #define PP2_TYPE_NOOP           0x04
        #define PP2_TYPE_UNIQUE_ID      0x05
        #define PP2_TYPE_SSL            0x20
        #define PP2_SUBTYPE_SSL_VERSION 0x21
        #define PP2_SUBTYPE_SSL_CN      0x22
        #define PP2_SUBTYPE_SSL_CIPHER  0x23
        #define PP2_SUBTYPE_SSL_SIG_ALG 0x24
        #define PP2_SUBTYPE_SSL_KEY_ALG 0x25
        #define PP2_TYPE_NETNS          0x30